.github/buildomat/build-and-test.sh

```diff
@@ -112,7 +112,17 @@ export RUSTC_BOOTSTRAP=1
 
 # Build all the packages and tests, and keep track of how long each took to build.
 # We report build progress to stderr, and the "--timings=json" output goes to stdout.
-ptime -m cargo build -Z unstable-options --timings=json --workspace --tests --locked --verbose 1> "$OUTPUT_DIR/crate-build-timings.json"
+#
+# The build graph ends up building several bin/test targets that depend on
+# omicron-nexus at the same time, which uses significant memory to compile on
+# illumos. To mitigate this we build everything except omicron-nexus's bin/test
```

Comment:
Nice win!

Is there something illumos-specific about this?

Does this suggest an upstream `cargo` bug?

Reply:
Also: if this is a memory problem, should we consider bumping the size of the buildomat runners instead?

Reply:
It's not entirely illumos-specific, although illumos does appear to use more (maybe 10-20%?) more memory than Ubuntu in the same stage of compilation. I don't think anything necessarily suggests a rustc or cargo bug yet; I have a hunch that our memory woes go hand-in-hand with compile time woes and probably have similar root causes. Every time we go down that path it feels like we're chasing dead ends.

@jclulow and I discussed bumping the builder size but we certainly don't want to do it for all Buildomat jobs; some manner of indicating that we have an XL-sized job would be necessary. I suspect even with this hack in place we'll push up against memory limits again.